File: answers/leetcode/Game of Life/Game of Life.java
class Solution {
    public void gameOfLife(int[][] board) {
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[i].length; j++) {
                int count = (i > 0 && j > 0 && board[i - 1][j - 1] > 0 ? 1 : 0)
                    + (i > 0 && board[i - 1][j] > 0 ? 1 : 0)
                    + (i > 0 && j + 1 < board[i].length && board[i - 1][j + 1] > 0 ? 1 : 0)
                    + (j > 0 && board[i][j - 1] > 0 ? 1 : 0)
                    + (i + 1 < board.length && j > 0 && board[i + 1][j - 1] > 0 ? 1 : 0)
                    + (j + 1 < board[i].length && board[i][j + 1] > 0 ? 1 : 0)
                    + (i + 1 < board.length && j + 1 < board[i].length && board[i + 1][j + 1] > 0 ? 1 : 0)
                    + (i + 1 < board.length && board[i + 1][j] > 0 ? 1 : 0);
                if (board[i][j] == 1 && count < 2) {
                    board[i][j] = 2;
                } else if (board[i][j] == 1 && count > 3) {
                    board[i][j] = 2;
                } else if (board[i][j] == 0 && count == 3) {
                    board[i][j] = -1;
                }
            }
        }
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[i].length; j++) {
                if (board[i][j] == 2) {
                    board[i][j] = 0;
                } else if (board[i][j] == -1) {
                    board[i][j] = 1;
                }
            }
        }
    }
}

File: answers/leetcode/Game of Life/Game of Life.cpp
//@type Array
//@result 22 / 22 test cases passed. Status: Accepted Runtime: 0 ms Submitted: 0 minutes ago

class Solution {
public:
    void gameOfLife(vector<vector<int>>& board) {
		if (0 == board.size()) {
			return;
		}
		vector<vector<int> > count_mat(board.size(), vector<int> (board[0].size(), 0));
        for (size_t i = 0; i < board.size(); ++ i) {
			for (size_t j = 0; j < board[i].size(); ++ j) {
				if (1 == board[i][j]) {
					for (int k = -1; k <= 1; ++ k) {
						for (int m = -1; m <= 1; ++ m) {
							if (i + k >= 0 && i + k < board.size() && j + m >= 0 && j + m < board[i + k].size() && (k | m)) {
								++ count_mat[i + k][j + m];
							}
						}
					}
				}
			}
		}
        for (size_t i = 0; i < count_mat.size(); ++ i) {
			for (size_t j = 0; j < count_mat[i].size(); ++ j) {
				if (count_mat[i][j] < 2 || count_mat[i][j] > 3) {
					board[i][j] = 0;
				}
				if (3 == count_mat[i][j]) {
					board[i][j] = 1;
				}
			}
		}
    }
};